### PR TITLE
Add ACIS thermal check tools and supporting package to skare

### PR DESCRIPTION
**Pull request by jeanconn**
Add ACIS thermal check tools and supporting package to skare

### CODE REVIEW

**Comment by jeanconn** [comment]
Packages that I made are in the expected location in /proj/sot/ska/pkgs .
For install, I think we may want to update the launchers in /proj/sot/ska/bin instead of just removing them.  The new ones would just look like starcheck:
```
fido: more /proj/sot/ska/bin/starcheck
 #!/bin/bash

SKA_bin=`dirname $0`
eval `${SKA_bin}/flt_envs -shell sh`
exec ${SKA_ARCH_OS}/bin/starcheck "$@"
```
This gives you environment-independent launchers as needed, but I'm not actually sure how these tools are called/run from ACIS so that would be good to know. @jzuhone ?

For lightweight regression testing, I think that ska_testr is presently not much help, as it is set up to run /proj/sot/ska/share/dea/dea_check.py to see if changes in the environment have accidentally caused changes in the tool output (classic regression), but is not designed to do integration tests for this application (does the application as installed in skare produce the intended output).

**Comment by jeanconn** [comment]
I also don't expect to be on LR tomorrow, so it would be good to update this PR with any notes about promotion that come up in that discussion.

**Comment by jeanconn** [comment]
I was thinking that the install strategy in HEAD flight 64 would involve:

   - installing the updated packages
   - moving the current /proj/sot/ska/bin launchers to *.old or some such
   - putting in new launchers in /proj/sot/ska/bin (if we decide we want them)
   - having ACIS do a run test of each in the installed environment.

**Comment by jzuhone** [comment]
Hi @jeanconn, I apologize for the delay in responding, today has been a bit hectic. 

The only issue I see with this is that the `dea_check`, etc are Python packages which install the `dea_check` script automatically to `bin` in the Python root. They are designed to be installed with `python setup.py install`. So how would this work with what you laid out above? 

Do you need anything else from me?

**Comment by jeanconn** [comment]
@jzuhone No need to apologize for a delay when I opened this PR 20 mins ago :smile:.

We pip install (via https://github.com/sot/skare/blob/master/cfg/python_modules.cfg#L23)  and in my tests of installing these new packages into a dev ska, your scripts are, appropriately, created in SKA_ARCH_OS/bin (our Python root in ).  So that's fine.   The question I mostly have for you is how do you run them in your ACIS LR process? Do you source into a ska env and then just call dea_check?  If so, I don't need to worry about launchers in /proj/sot/ska/bin because you'll get dea_check in the path in /proj/sot/ska/arch/x86_64-linux_CentOS-6/bin (SKA_ARCH_OS/bin).  If your scripts already call /proj/sot/ska/bin/dea_check, I'll need to update that wrapper.  If your scripts do something like call /proj/sot/ska/bin/python /proj/sot/ska/share/dea_check/dea_check.py then your scripts will need update.

**Comment by jzuhone** [comment]
Our thermal models use `/proj/sot/ska/bin/dea_check`, etc. 

So the reason for the wrapper is that it points to the appropriate file for the correct arch?

**Comment by jeanconn** [comment]
Somewhat.

Using /proj/sot/ska/bin/dea_check as an example, the bigger deal is that
the wrapper runs that flt_envs script to source into a full ska environment
(which is what the /proj/sot/ska/bin/dea_check is already doing).  So
that's why we don't want to just, say, symlink to your entrypoint script.

Regarding the appropriate file for the correct arch, our multi-arch support
isn't something that you really need (because this doesn't really need to
run on anything but linux 64), but since we need to replace
/proj/sot/ska/bin/dea_check anyway we might as well use the both general
and correct form with that SKA_ARCH_OS path.

**Comment by taldcroft** [comment]
I think the solution depends a bit on the philosophy (and practice) for actually running the load review checking scripts.  

Currently the idea is that you DO NOT need to be in the Ska environment to run the script, and so one explicitly calls the launcher script /proj/sot/ska/bin/dea_check to do all the setup and insist on the Flight environment.  This has some heritage from use of starcheck within the mission planning environment where they did not always get things right.

But you can get the same result (with multi-arch support), by running an unqualified `dea_check.py` and relying on the path to find it.  For a controlled situation like running within ACIS "LR" this is possible and can be reliable, and eliminates the need for separate launcher scripts.  This is really the intention for entry point scripts, so the only question is whether we are comfortable in terms of possibly running the wrong `dea_check.py` in the path.

**Comment by jeanconn** [comment]
Clarification, do you mean `dea_check.py` above or the `dea_check` entry point script?

**Comment by jzuhone** [comment]
@taldcroft—just for clarification `dea_check.py` is no longer exposed as something that can be run as in `python dea_check.py`. It’s internal to the package. The `dea_check` run script is the only entry point to the tool.

**Comment by taldcroft** [comment]
I meant `dea_check` and have edited accordingly.  (I was asking myself about the actual new name of the entry point script but didn't take the time to check).

**Comment by jeanconn** [comment]
Regarding  "so the only question is whether we are comfortable in terms of possibly running the wrong dea_check.py in the path.", I think there are two questions:

1) Do we want to make that change (to using `dea_check` in the path instead of a SKA/bin launcher with environment settings)?
2) Do we want to make that change on this installation?

**Comment by taldcroft** [comment]
My recommendation is that we should move toward using entry point scripts as the generic way of using scripts for Ska-based tools.  This reduces the application footprint on the Ska runtime environment since now we need "out-of-Ska" launcher scripts for each flight-configured script.

It seems we can do this reliably within an application like LR by using one of `/proj/sot/ska/bin/skare` (generic Ska launcher) or `/proj/sot/ska/bin/python` (Ska python launcher).  E.g. from perl you can do something like below and be guaranteed of running `$SKA_ARCH_OS/bin/dea_check` from within the Ska environment (with Ska first in the path):
```
@cmd = ("/proj/sot/ska/bin/python", "dea_check")
system(@cmd)
```
or
```
@cmd = ("/proj/sot/ska/bin/skare", "python", "dea_check")
..
```
(BTW I'm assuming that the thermal check tools are run automatically by LR or some-such.)

So yes, the idea is to do this change for the new suite of ACIS thermal check tools.  In fact this solves the fall-back problem since the old versions can remain installed in `/proj/sot/ska/share` for now, and once soak is done (after a few loads) we can remove them.

Does that all hang together?  Have I missed anything?

**Comment by jzuhone** [comment]
Yeah, I think this works.

**Comment by jeanconn** [comment]
>  In fact this solves the fall-back problem since the old versions can remain installed in /proj/sot/ska/share for now, and once soak is done (after a few loads) we can remove them.

I'm not quite following that, as my understanding was that there was no conflict and no fall-back problem.

**Comment by jeanconn** [comment]
Regarding:

> (BTW I'm assuming that the thermal check tools are run automatically by LR or some-such.)

Do we want to make a note of what is actually planned for changes to the ACIS LR calls somewhere then (in this PR or someplace else)?

**Comment by jzuhone** [comment]
So here's my question... where does the entry point script `dea_check` get placed once the package is pip-installed? The `setup.py` for that package (and the rest) is designed to automatically put it in the `bin` directory of the Python stack you're using.

**Comment by jeanconn** [comment]
This makes my new install process:

   - install the updated packages
   - move the current /proj/sot/ska/bin launchers to *.old or some such
   - confirm update of any ACIS calling code
   - have ACIS do a run test of each check tool in the installed environment.

**Comment by jeanconn** [comment]
Sorry, missed your question while typing @jzuhone .  The new `dea_check` entry point script is in /proj/sot/ska/arch/x86_64-linux_CentOS-6/bin`, but per Tom's comments about possible ways to configure your calling code you shouldn't need to care.  It is just in the SKA path.

**Comment by jzuhone** [comment]
@jeanconn thanks, that's what I figured, but wanted to make sure.

**Comment by taldcroft** [comment]
@jeanconn - your process looks good.  I would just add a step after installing the updated packages to try running the legacy packages (at least one of them) to explicitly verify that fallback does not require un-installing the new packages.

**Comment by jeanconn** [comment]
@taldcroft Do you want that verified in a test ska before we go with this (I don't have the legacy packages presently set up in one but can do that too if desired).

**Comment by taldcroft** [comment]
> Do you want that verified in a test ska before we go with this

No.  In fact just belay that whole idea.  `pip uninstall` really does work, so in the very unlikely event we have to fall back we'll just uninstall the new packages and remove the `.old` suffixes and it will be done.

**Comment by jeanconn** [comment]
For the record, for install on 15-Dec-2017 we're going to go back to the plan of updating the wrappers in SKA/bin, with a plan to remove those at a future point with an accompanying update to the ACIS LR script.

File: pkgs.manifest [hunk 1]
```diff
@@ -44,6 +44,12 @@ sherpa-4.7.tar.gz
 ska_path-3.1.tar.gz
 starcheck-11.22.tar.gz
 #
+# ACIS ops packages
```
Comment (jzuhone) [inline pkgs.manifest]: Hi @jeanconn--one minor thing left here. I've merged everything and made the tags, but I've also made all of the version numbers consistent in terms of having `major.minor.patch` format. 

So the versions you should use are:

`dpa_check`: `2.0.0`
`dea_check`: `2.0.0`
`psmc_check`: `1.0.0`
`acis_thermal_check`: `2.0.0`

Comment (jeanconn) [reply pkgs.manifest]: Thanks for the heads-up @jzuhone .  And yes, I saw these when I was making the final tar gz from the checked-out releases.  All looks good.

**Pull request merged**